source/common/common/token_bucket_impl.cc

```diff
@@ -103,4 +107,18 @@ double AtomicTokenBucketImpl::timeNowInSeconds() const {
   return std::chrono::duration<double>(time_source_.monotonicTime().time_since_epoch()).count();
 }
 
+std::chrono::milliseconds AtomicTokenBucketImpl::nextTokenAvailable() const {
+  // If there are tokens available, return immediately.
+  if (remainingTokens() >= 1) {
+    return std::chrono::milliseconds(0);
+  }
+
+  // Calculate time since the last fill.
+  double current_time = timeNowInSeconds();
+  double time_since_last_fill = std::fmod(current_time, fill_interval_);
+  double time_until_next_fill = fill_interval_ - time_since_last_fill;
+  return std::chrono::milliseconds(
+      static_cast<uint64_t>(time_until_next_fill * MILLISECONDS_PER_SECOND));
+}
```

Comment:
Is this make actual sense by calculate the `std::fmod`? The local rate limit filter won't use fix time windows anyway?

Reply:
/wait-any

Reply:
And in most cases, then fill_rate_ will be a value > 10 or > 100, that means in most cases, the nextTokenAvailable() will always less then 1s and your `x-local-rate-limit` will always be zero.

Reply:
Updated

> And in most cases, then fill_rate_ will be a value > 10 or > 100, that means in most cases, the nextTokenAvailable() will always less then 1s and your x-local-rate-limit will always be zero.

TBH, that's fine to me, because of we use int as return values(same as ratelimit filter).